File: app/src/main/java/space/pal/sig/model/database/converters/NewsSourceConverter.java
package space.pal.sig.model.database.converters;

import androidx.room.TypeConverter;

import space.pal.sig.model.NewsSource;

/**
 * SpaceNow
 * Created by Catalin on 7/19/2020
 **/
public class NewsSourceConverter {

    @TypeConverter
    public static NewsSource fromString(String source) {
        return source == null ? null : NewsSource.valueOf(source);
    }

    @TypeConverter
    public static String toString(NewsSource source) {
        return source == null ? null : source.name();
    }

}

File: app/src/main/java/space/pal/sig/database/converters/NewsSourceConverter.kt
package space.pal.sig.database.converters

import androidx.room.TypeConverter
import space.pal.sig.model.entity.NewsSource

/**
 * SpaceNow
 * Created by Catalin on 12/24/2020
 **/
class NewsSourceConverter {

    @TypeConverter
    fun fromString(string: String): NewsSource {
        return NewsSource.fromText(string)
    }

    @TypeConverter
    fun toString(obj: NewsSource): String {
        return obj.text
    }

}